### PR TITLE
Add basic recycler implementation for 1.0

### PR DESCRIPTION
**Pull request by jcmoore**
Now would be a great time to comment on names/APIs @dlasky / @anthonykoerber / @shuwen

### CODE REVIEW

File: src/shared/behaviors/domwatchable.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<!--
```
Comment (dlasky) [inline src/shared/behaviors/domwatchable.html]: this seems like an exact duplicate of dommutable only with less safety around node removal

File: src/mm-item-recycler/mm-item-recycler.js [hunk 2]
```diff
@@ -15,165 +15,151 @@ subject to an additional IP rights grant found at http://polymer.github.io/PATEN
...
-					while (unusable-- > 0) {
-						if (head._virtualPanelIndex < 1) {
-							break;
-						}
+		_scrollTop: 0,
+		extraItems: 5,
 
-						presentation.pop();
+		properties: {
+			ver: {
```
Comment (dlasky) [inline src/mm-item-recycler/mm-item-recycler.js]: ver is deprecated in 1.0 in favor of Strand.VERSION (tbd)

File: src/mm-item-recycler/mm-item-recycler.js [hunk 3]
```diff
@@ -15,165 +15,151 @@ subject to an additional IP rights grant found at http://polymer.github.io/PATEN
...
 
-				this.refresh(true, 0);
-			}
+		initialized: false,
+		itemHeight: 0,
+
+		_recycler: null,
+
+		ready: function () {
+			this._recycler = new Recycler(
```
Comment (dlasky) [inline src/mm-item-recycler/mm-item-recycler.js]: this should be defined in properties with a value function

File: src/mm-item-recycler/mm-item-recycler.js [hunk 4]
```diff
@@ -15,165 +15,151 @@ subject to an additional IP rights grant found at http://polymer.github.io/PATEN
...
+		},
 
-				}
+		_onFocus: function(e) {
+			this._updateScrollbarUI(0);
+		},
 
-				this.refresh(true, 0);
-			}
+		initialized: false,
```
Comment (dlasky) [inline src/mm-item-recycler/mm-item-recycler.js]: these should be defined in properties as well

File: src/mm-item-recycler/mm-item-recycler.js [hunk 5]
```diff
@@ -187,720 +173,219 @@ found here: https://github.com/Polymer/core-list
...
 				this.initializeRecycler();
 			}
-
-			this.dataLength = this.data.length;
 		},
 
-		initializeTemplateBind: function() {
-			if(typeof this.itemTemplate === "string") {
-				this.itemTemplate = this.querySelector("#" + this.itemTemplate);
+		itemTemplateElement: null,
```
Comment (dlasky) [inline src/mm-item-recycler/mm-item-recycler.js]: props

**Comment by dlasky** [comment]
@jcmoore this is old now right? can we close it?

**Comment by jcmoore** [comment]
yep

**Pull request closed without merging**